### DISCUSSION TITLE
MsQUIC Direct Linux APIs in the API document

### DISCUSSION THREAD

**Discussion by sanjayv161** [microsoft/msquic#4033] [Q&A]
It would greatly benefit the developer community, particularly those accustomed to Linux environments, if the msquic API documentation included sections specifically detailing the use of key APIs like Register, Create Configuration, and Listener in a Linux context. As many developers may not be familiar with Windows-based programming, having Linux-specific examples and explanations would significantly ease the learning curve and enhance the accessibility of msquic for a broader range of users. Incorporating these changes into the documentation would make msquic more approachable and user-friendly for developers primarily working in Linux environments.

**Comment by nibanks** [comment]
Thanks for your post! We have some high-level API documentation [here](https://github.com/microsoft/msquic/blob/main/docs/API.md). While not 100% complete, it should give you a general overview. That being said, do you have any specific things you'd like to see documented? Additionally, the API is meant to be cross-platform and generally shouldn't have any platform/OS specific constructs (Windows or Linux). So I'm also curious as to what you see as Windows specific.
